### PR TITLE
RANCHER-1779: deployModuleFromFeatureBranch Pipeline should support GRADLE builds

### PR DESCRIPTION
**Pull request by yaroslavishchenko**
Related Jira Ticket:
[RANCHER-1779: deployModuleFromFeatureBranch Pipeline should support GRADLE builds](https://folio-org.atlassian.net/browse/RANCHER-1779)

Pipeline with verification results:
https://jenkins-aws.indexdata.com/job/folioRancher/job/tmpFolderForDraftPipelines/job/YaroslavI/job/deployEurekaModuleFromFeatureBranch-1779/

### CODE REVIEW

**Review by aatoyan — APPROVED** [review]
If it works looks good for me :)

File: pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                              [$class: 'SubmoduleOption', recursiveSubmodules: true]],
+          userRemoteConfigs: [[url: "https://github.com/folio-org/${module.name}.git"]]
+        ])
+      }
+
+      stage('[Maven/Gradle] Compile') {
+        // Check if Maven or Gradle build tool is used
+        module.buildTool = fileExists("${module.name}/pom.xml") ? 'maven' : 'gradle'
+
+        if (module.buildTool == 'maven') {
```
Comment (OHaimanov) [inline pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile:92]: It is will looks better if change 'if' to swtich/case and also move common(repeatable) code to separate function. Even inside this Jenkinsfile

File: pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            // Load Module Details for Gradle builds
+            module.loadModuleDetails("${module.name}-${module.version}.${featureHash}")
+
+            // Build Module as Gradle Project
+            sh "./gradlew ${commonGradleOpts} -PappVersion=${module.version} assemble"
+          }
+        }
+      }
+
+      stage('[Docker] Build and push') {
```
Comment (OHaimanov) [inline pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile:134]: Same here, it's lot of code duplicates please refactor/optimise

File: pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile (hunk 1 above)
Comment (yaroslavishchenko) [reply pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile]: Code adjusted accordingly to your suggestion.
Please take a look on changes.

File: pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile (hunk 2 above)
Comment (yaroslavishchenko) [reply pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile:140]: Code adjusted accordingly to your suggestion.
Please take a look on changes.

**Comment by yaroslavishchenko** [comment]
Pay attention for comment from #634 pull request:


> - There is not logic for handling special case for mod-consortia-keycloak.
> - Newly generated app should be entitled ONLY on top of requested tenant, not all tenants available on env.
> - As was mentioned on one of daily meeting, if entitlement has failed, it should be cleaning up of pre-entitlement resources (registered apps, modules discovery and so on)
> - Additional improvements are required in the logic for updating apps.

**Pull request merged**